src/uproot/interpretation/library.py

```diff
@@ -840,105 +839,22 @@ def imported(self):
 
     def finalize(self, array, branch, interpretation, entry_start, entry_stop):
         pandas = self.imported
+        index = _pandas_basic_index(pandas, entry_start, entry_stop)
 
-        if isinstance(array, uproot.interpretation.objects.StridedObjectArray):
-            arrays = []
-            columns = []
-            _strided_to_pandas((), array.interpretation, array.array, arrays, columns)
-            maxlen = max(len(x) for x in columns)
-            if maxlen == 1:
-                columns = [x[0] for x in columns]
-            else:
-                columns = pandas.MultiIndex.from_tuples(
-                    [x + ("",) * (maxlen - len(x)) for x in columns]
-                )
-            index = _pandas_basic_index(pandas, entry_start, entry_stop)
-            out = pandas.DataFrame(
-                dict(zip(columns, arrays)), columns=columns, index=index
-            )
-            out.leaflist = maxlen != 1
-            return out
-
-        elif isinstance(array, uproot.interpretation.jagged.JaggedArray) and isinstance(
-            array.content, uproot.interpretation.objects.StridedObjectArray
+        if (
+            (not (isinstance(array, numpy.ndarray)))
+            or (array.dtype.names is not None)
+            or (len(array.shape) != 1)
```

Comment:
Unnecessary parentheses.

```suggestion
            not (isinstance(array, numpy.ndarray))
            or array.dtype.names is not None
            or len(array.shape) != 1
```

But also, this `if`-`then`-`else` is a little hard to read because it's defined in terms of a negative. The simple case is in the `else` block and the `if` predicate is "not the simple case."

It would be more readable like this:

```python
if (
    isinstance(array, numpy.ndarray)
    and array.dtype.names is None
    and len(array.shape) == 1
):
    return pandas.Series(array, index=index)
else:
    # more complicated thing with awkward-pandas
```

(Fixing that would override the suggestion I made above. But still, you don't need the extra parentheses.)

Reply:
Changed as suggested 👍🏻